File: Lecture 20 Solving LeetCodeCodeStudio Questions [Arrays]/Move Zeroes to End of Array.cpp
#include <iostream>
#include<bits/stdc++.h>

using namespace std;
//Program to place all zeros to end of array
void moveZeroes(int nums[], int n) {
    int i = 0;
    
    for(int j=0; j< n; j++)
    {
        if(nums[j]!=0)
        {
            swap(nums[j], nums[i]);
            i++;
        }
    }
}
    
int main()
{
    int arr[] = {0,1,0,3,12};
    int n = 5;
    moveZeroes(arr,n);
	
	for(int i=0; i<n; i++)
	{
	    cout<<arr[i]<<" ";
	}
}

File: Lecture 20 Solving LeetCodeCodeStudio Questions [Arrays]/Move Zeroes to End of Array.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.*;
public class Main{
    //Program to place all zeros to end of array
    public static void moveZeroes(int nums[], int n) 
    {
        int i = 0;
        
        for(int j = 0; j < n; j++)
        {
            if(nums[j]!=0)
            {
                int temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;
                i++;
            }
        }
    }
    
	public static void main(String[] args) {
	    int arr[] = {0,1,0,3,12};
        int n = 5;
        moveZeroes(arr,n);
    	
    	for(int i=0; i<n; i++)
    	{
    	    System.out.println(arr[i]);
    	}
	}
}
